src/gui/operations/RenameOperation.cpp

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Damir Porobic <damir.porobic@gmx.com>
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor,
+ * Boston, MA 02110-1301, USA.
+ */
+
+#include "RenameOperation.h"
+
+#include <utility>
+
+RenameOperation::RenameOperation(QWidget *parent, const QString &pathToImageSource, const QString &imageFilename, IToastService *toastService)
+	: mParent(parent), mPathToImageSource(pathToImageSource), mImageFilename(imageFilename), mToastService(toastService)
+{
+}
+
+SaveResultDto RenameOperation::execute()
```

Comment:
The SaveResultDto is kind of misleading here, I had to check the code to see if you are saving something but you're actually just renaming. How about introducing a new RenameResult Object that is only used for this purpose? It could have just the current or old name and a bool flag that tells us if the rename was successful or not?